classes/03class/exercises/c03-aws03/alanlima/README.md

```diff
@@ -0,0 +1,135 @@
+            {
+                "Action": "UPSERT",
+                "ResourceRecordSet": {
+                    "Name": "c03aws03.devops.rocks",
+                    "Type": "CNAME",
+                    "TTL": 600,
+                    "ResourceRecords": [
+                        {
+                            "Value": "devopsacademy-lb-1691285263.ap-southeast-2.elb.amazonaws.com"
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+    ```
+    
+    * Using UPSERT because it is practical and I may need to update it anyway
+
+- Checking that it actually works..
+
+    ```bash
+    $ ssh ec2-user@13.236.232.76
+    Last login: Thu Jul  9 08:57:30 2020 from 122-148-135-233.sta.wbroadband.net.au
+    
+           __|  __|_  )
+           _|  (     /   Amazon Linux 2 AMI
+          ___|\___|___|
+    
+    https://aws.amazon.com/amazon-linux-2/
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-104.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    
+    [ec2-user@ip-172-16-10-104 ~]$ curl c03aws03.devops.rocks
+    ip-172-16-10-117.ap-southeast-2.compute.internal
+    ```
+
+- Post any findings, how option you decided to go with, records created, domain created, and anything else that you find interesting.
+
+    - I was going to do the public hosted zone, but it took a while until I receive the e-mail for the free domain registration. Then, I started the private, make it works, got excited, received a verification e-mail, did the public zone as well 😅
+    - AWS will set the hosted zone to private when VPC is specified, even if you set PrivateZone=False. I needed to recreate the route a couple of times until I release that. The CLI didn't give any warning. 😢
+    - I found the docs to update the recordset not as clear as the other docs. I could not find exactly how I should create the A record to link to the ALB. I ended up creating it on the console and listed the resources to check how it was created.
+    - I would expect some small delay until the route 53 propagate the changes, but as soon as I set it, I was able to connect to my private hosted zone. Impressive 👌👌.
+    - Overall, I found it quite easy to set up a private hosted zone. When I started, I thought that I would need to execute a thousand commands.
+    - Forgot to set the key pair to my instances so I needed to recreate the launch configuration, ASG and update the LB register again. 🙄 *I am getting good in recreating things 🤣🤣*
```

Comment:
experience! 😆 

Interesting that now ASG supports `launch templates` (along with `launch configurations`). Launch templates are user for a long time for EC2 instances, it is an easy way to template your instance and launch it whenever you want.  Now you can reuse the same template for either EC2 instances or ASGs.

Reply:
Interesting, I am going to research more about it. I actually was creating some copy with original launch configurations for some testings and I was renaming it to v2, v3, and so on. Launch templates has versions control build-in. Cool 🤗🤗